wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectApiModule.java

```diff
@@ -92,6 +101,38 @@ protected void configure() {
         configureVfsFilters(excludeMatcher);
         configureVfsFilters(fileWatcherExcludes);
         configureVfsEvent();
+        configureTreeWalker();
+    }
+
+    private void configureTreeWalker() {
```

Comment:
Can you create single guice  module in che-core-api-project/src/main/java/org/eclipse/che/api/vfs/watcher/